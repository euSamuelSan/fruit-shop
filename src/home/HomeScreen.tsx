import React from 'react'
import Shop from '../shop'
import HeaderComponent from '../header/HeaderComponent'
import Footer from '../footer/FooterComponent'

export default function HomeScreen(): JSX.Element {
    return (
        <>
            <HeaderComponent />
            <Shop />
            <Footer />
        </>
    )
}
