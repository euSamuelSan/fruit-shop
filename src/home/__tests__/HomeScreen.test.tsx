import { render, RenderResult } from '@testing-library/react'

import HomeScreen from '../HomeScreen'

describe('HomeScreen', () => {
    const renderHomeScreen = (): RenderResult => {
        return render(<HomeScreen />)
    }

    it('Deve desenhar o componente corretamente', () => {
        renderHomeScreen()

        expect(document.body).toMatchSnapshot()
    })
})
