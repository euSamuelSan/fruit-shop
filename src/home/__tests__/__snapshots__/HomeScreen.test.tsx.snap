// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Deve desenhar o componente corretamente 1`] = `
<body>
  <div>
    <div
      class="sc-iTONeN hVhjZO"
    >
      <svg
        class="sc-iAvgwm hbzhyg"
        color="#ffffff"
        fill="currentColor"
        height="40"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 512 512"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M321.89 171.42C233 114 141 155.22 56 65.22c-19.8-21-8.3 235.5 98.1 332.7 77.79 71 197.9 63.08 238.4-5.92s18.28-163.17-70.61-220.58zM173 253c86 81 175 129 292 147"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="32"
        />
      </svg>
      <h1
        class="sc-efBctP fWfCFy"
      >
        Sua Horta
      </h1>
      <div
        class="sc-cTQhss cmwdyS"
      >
        <button
          class="sc-jOrMOR jsAYQy"
        >
          <svg
            color="#ffffff"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Carrinho
            </title>
            <path
              d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
            />
          </svg>
        </button>
        <button
          class="sc-jOrMOR jsAYQy"
        >
          <svg
            color="#ffffff"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Sair
            </title>
            <path
              d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 0 1-112.7 75.9A352.8 352.8 0 0 1 512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 0 1-112.7-75.9 353.28 353.28 0 0 1-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 0 0 0-12.6z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="sc-breuTD gRSwPg"
    >
      <div
        class="sc-ksZaOG hWsKsx"
      >
        <div
          class="sc-gsnTZi oaevL"
        >
          <svg
            class="sc-dkzDqf gavTNQ"
            color="#8e8e93"
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(142, 142, 147);"
            viewBox="0 0 1024 1024"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
          <input
            class="sc-bczRLJ hKuxkL sc-hKMtZM bTFJgM"
            placeholder="Pesquise por nome do produto"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="sc-iBkjds hMNVUq"
      />
    </div>
    <div
      class="sc-cOFTSb huTOMH"
    >
      <p
        class="sc-BeQoi fVnnro"
      >
        Â© 2022 SAN, Samuel.
      </p>
      <div
        class="sc-kgUAyh bEGnhB"
      >
        <a
          class="sc-hTtwUo dhNByH"
          href="https://github.com/euSamuelSan"
          target="_blank"
        >
          Github 
          <svg
            color="#ffffff"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
            />
          </svg>
        </a>
        <a
          class="sc-hTtwUo dhNByH"
          href="https://www.linkedin.com/in/samuel-san-juan/"
          target="_blank"
        >
          LinkedIn 
          <svg
            color="#ffffff"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M847.7 112H176.3c-35.5 0-64.3 28.8-64.3 64.3v671.4c0 35.5 28.8 64.3 64.3 64.3h671.4c35.5 0 64.3-28.8 64.3-64.3V176.3c0-35.5-28.8-64.3-64.3-64.3zm0 736c-447.8-.1-671.7-.2-671.7-.3.1-447.8.2-671.7.3-671.7 447.8.1 671.7.2 671.7.3-.1 447.8-.2 671.7-.3 671.7zM230.6 411.9h118.7v381.8H230.6zm59.4-52.2c37.9 0 68.8-30.8 68.8-68.8a68.8 68.8 0 1 0-137.6 0c-.1 38 30.7 68.8 68.8 68.8zm252.3 245.1c0-49.8 9.5-98 71.2-98 60.8 0 61.7 56.9 61.7 101.2v185.7h118.6V584.3c0-102.8-22.2-181.9-142.3-181.9-57.7 0-96.4 31.7-112.3 61.7h-1.6v-52.2H423.7v381.8h118.6V604.8z"
            />
          </svg>
        </a>
        <a
          class="sc-hTtwUo dhNByH"
          href="https://instagram.com/euSamuelSan"
          target="_blank"
        >
          Instagram 
          <svg
            color="#ffffff"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 306.9c-113.5 0-205.1 91.6-205.1 205.1S398.5 717.1 512 717.1 717.1 625.5 717.1 512 625.5 306.9 512 306.9zm0 338.4c-73.4 0-133.3-59.9-133.3-133.3S438.6 378.7 512 378.7 645.3 438.6 645.3 512 585.4 645.3 512 645.3zm213.5-394.6c-26.5 0-47.9 21.4-47.9 47.9s21.4 47.9 47.9 47.9 47.9-21.3 47.9-47.9a47.84 47.84 0 0 0-47.9-47.9zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zm-88 235.8c-7.3 18.2-16.1 31.8-30.2 45.8-14.1 14.1-27.6 22.9-45.8 30.2C695.2 844.7 570.3 840 512 840c-58.3 0-183.3 4.7-235.9-16.1-18.2-7.3-31.8-16.1-45.8-30.2-14.1-14.1-22.9-27.6-30.2-45.8C179.3 695.2 184 570.3 184 512c0-58.3-4.7-183.3 16.1-235.9 7.3-18.2 16.1-31.8 30.2-45.8s27.6-22.9 45.8-30.2C328.7 179.3 453.7 184 512 184s183.3-4.7 235.9 16.1c18.2 7.3 31.8 16.1 45.8 30.2 14.1 14.1 22.9 27.6 30.2 45.8C844.7 328.7 840 453.7 840 512c0 58.3 4.7 183.2-16.2 235.8z"
            />
          </svg>
        </a>
        <a
          class="sc-hTtwUo dhNByH"
          href="https://twitter.com/euSamuelSan"
          target="_blank"
        >
          Twitter 
          <svg
            color="#ffffff"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0 0 75-94 336.64 336.64 0 0 1-108.2 41.2A170.1 170.1 0 0 0 672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 0 0-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 0 1-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 0 1-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</body>
`;
